main.cc

```diff
@@ -446,6 +446,19 @@ class cql_config_updater {
     }
 };
 
+template <typename Func>
+inline auto defer_with_log_on_error(Func&& func) {
+    auto func_with_log = [func = std::forward<Func>(func)] () mutable {
+        try {
+            std::forward<Func>(func)();
```

Comment:
maybe I'm missing something, but what does this std::forward do? Wouldn't a simple func() work?